verify/verify.go

```diff
@@ -886,23 +909,33 @@ func GetAttestationFromReport(report *spb.Report, options *Options) (*spb.Attest
 // on the report's SignatureAlgo and uses the AMD Key Distribution Service to download the
 // report's corresponding VCEK certificate.
 func SnpReport(report *spb.Report, options *Options) error {
+	return SnpReportContext(context.Background(), report, options)
+}
+
+// SnpReportContext behaves like SnpReport but forwards the context to the HTTPSGetter.
+func SnpReportContext(ctx context.Context, report *spb.Report, options *Options) error {
 	if options.DisableCertFetching {
 		return errors.New("cannot verify attestation report without fetching certificates")
 	}
-	attestation, err := GetAttestationFromReport(report, options)
+	attestation, err := GetAttestationFromReportContext(ctx, report, options)
 	if err != nil {
 		return fmt.Errorf("could not recreate attestation from report: %w", err)
 	}
-	return SnpAttestation(attestation, options)
+	return SnpAttestationContext(ctx, attestation, options)
 }
 
 // RawSnpReport verifies the raw bytes representation of an attestation report's signature
 // based on the report's SignatureAlgo and uses the AMD Key Distribution Service to download
 // the report's corresponding VCEK certificate.
 func RawSnpReport(rawReport []byte, options *Options) error {
+	return RawSnpReportContext(context.Background(), rawReport, options)
+}
+
+// RawSnpReportContext behaves like RawSnpReport but forwards the context to the HTTPSGetter.
+func RawSnpReportContext(ctx context.Context, rawReport []byte, options *Options) error {
```

Comment:
Does it make sense to add a context version of this convenience function, or should the caller just inline it if needed?

Reply:
It doesn't hurt.

Reply:
Ok, then let's keep it.